addon/debounce-task.ts

```diff
@@ -136,29 +136,28 @@ export function cancelDebounce(
   obj: EmberObject,
   name: string
 ): void | undefined {
-  let pendingDebounces: Object = registeredDebounces.get(obj);
-
-  if (pendingDebounces === undefined || pendingDebounces[name] === undefined) {
+  if (!registeredDebounces.has(obj)) {
     return;
   }
+  const pendingDebounces: PendingDebounces = registeredDebounces.get(obj);
 
-  let { cancelId } = pendingDebounces[name];
+  if (!pendingDebounces.has(name)) {
+    return;
+  }
+  // @ts-ignore
```

Comment:
Because `pendingDebounces.get(name)` returns `PendingDebounce | undefined`. The flow analysis does not understand `registeredDebounces.has(obj)` as a type guard: https://github.com/Microsoft/TypeScript/issues/13086

Reply:
Wouldn't this be a good candidate for `!`? Basically `registeredDebounces.get(obj)!` (which would instruct TS that it _for sure_ is not undefined)...

Reply:
Of course. I somehow forgot about its existence. 😅

Reply:
Good catch!